test/subdomain_test.coffee

```diff
@@ -0,0 +1,70 @@
+should = require('chai').Should()
+
+path = require 'path'
+
+
+mockery = require 'mockery'
+
+cozyInstanceMock =
+        getDomain: (callback) ->
+            callback null, "cozy.cozycloud.cc"
+        getLocale: (callback)->
+            callback null, "fr"
+
+
+applicationMock = domainSlug: (hostname, callback) ->
+        if hostname is "cozyblog.cozycloud.cc"
+            callback null, "blog"
+        else
+            callback null, ""
+
+mockery.registerMock path.join(__dirname, '../server/models/instance'), cozyInstanceMock
+mockery.registerMock path.join(__dirname, '../server/models/application'), applicationMock
```

Comment:
Lines should be less wide than 80 chars. See [Guidelines](https://github.com/cozy/cozy-guidelines#generic-code-style)